Done

ranelbe opened the pull request:
Done

ranelbe committed:
add a pptx parser feature

ranelbe committed:
add prompt composer feature

ranelbe committed:
improve the prompt suffix

ranelbe committed:
add openAI api manager feature

ranelbe committed:
Merge branch 'api-manager-feature' into dev

ranelbe committed:
add pptx summarize class

ranelbe committed:
add main.py including CLI

ranelbe committed:
Update README.md

ranelbe committed:
fix bug 'Event loop is closed' in main.py

ranelbe committed:
Merge branch 'dev' of https://github.com/Scaleup-Excellenteam/final-exercise-ranelbe into dev

ranelbe committed:
add requirements file

ranelbe committed:
add type hints to make the code more clear

ranelbe committed:
make some functions private

ranelbe committed:
add server+client functionality

ranelbe committed:
add tests

ranelbe committed:
add requirements.txt

excellenteam-scaleup commented on README.md:
```diff
@@ -1,5 +1,9 @@
 # The GPT-Explainer Project
 
+To run the script type:
```
Nicely explained

excellenteam-scaleup commented on api_manager.py:
Good logical separation of the API to a module.
Not sure we need a class for managing it, but it's a possible way for it

excellenteam-scaleup commented on api_manager.py:
```diff
@@ -0,0 +1,28 @@
+        """ initialization.
+        Set the API key from an environment variable.
+        """
+        openai.api_key = os.getenv("OPENAI_API_KEY")  # an environment variable
```
It would be better for the environment variable to be in a config file, or at least a constant and not magically appear here

excellenteam-scaleup commented on main.py:
```diff
@@ -0,0 +1,34 @@
+from pptx_summarizer import PPTXSummarizer
+
+
+def cli_configuration():
```
Nice implement of bonus requierments

excellenteam-scaleup commented on pptx_summarizer.py:
```diff
@@ -0,0 +1,42 @@
+from api_manager import APIManager
+import asyncio
+
+OUTPUT_FILE_NAME = 'result.json'
```
One of the requirements was for the json file to have the same name like the original pptx file

excellenteam-scaleup commented on pptx_parser.py:
Again - good separation of logical parts

excellenteam-scaleup commented on prompt_composer.py:
```diff
@@ -0,0 +1,26 @@
+
+PROMPT_SUFFIX = """provide a detailed explanation for this presentation slide, 
```
Good prompt

excellenteam-scaleup commented on requirements.txt:
Nice

excellenteam-scaleup commented on test.pptx:
It's nice that you added a test file. It would be nice if it also had some edge-cases like an empty slide

excellenteam-scaleup reviewed (approved):
Very good job!
Your git work looks great:
Commit messages are a little short, but explain the commit
Good logical division of branches, division to modules and files.
Your code is working, documented, follows PEP conventions, and divided into small simple functions.
note that when multiple pptx files are given the output needs to be saved in separate json files to keep track.
 
Keep up the good work!

ranelbe committed:
Add database for managing the data (use SQLite + SQLAlchemy).
Update the Explainer, WebAPI, and Client to establish connectivity with the database.

ranelbe committed:
Merge pull request #3 from Scaleup-Excellenteam/SQLAlchemy-feature

Add database for managing the data (use SQLite + SQLAlchemy).

ranelbe committed:
Update README.md

ranelbe committed:
Delete GPT-explainer-2-2.md

ranelbe committed:
Delete GPT-explainer.md
